### PR TITLE
Add tables to agent descriptions with CLI args

### LINKED ISSUE #1876: ParlAI auto-generated docs script for all standard agents
We should have a new section of the [docs](https://parl.ai/docs/) called Standard Agents. It should be a revision of the current "reference models" section (see attached).

It should use the docstrings for the agent module in order to populate this information.

It should additionally use `add_cmdline_args` to get a list of all the arguments this agent takes and display them. You may use [`generate_cli.py`](https://github.com/facebookresearch/ParlAI/blob/master/docs/source/generate_cli.py) as a template. Bonus points if you format as a pretty table though.

### PR DESCRIPTION
**Pull request by stephenroller**
**Patch description**
- Add tables showing the CLI args for the agents, going along with their READMEs.
- Move the logic that shows the "recommended" value in the command line `--help`. This enables me to also capture it cleanly at the web end, and will enable `no_save` and `no_override` later.
- Fix a few minor things about the documentation, some small bugs, etc.
- Make the build_website step actually fail on warnings, ensure we're aware some pretty regular issues that were cropping up.

Combined with #2286, this closes #1876.

**Testing**
Screen Shot 2020-01-18 at 12 04 45 AM

### CODE REVIEW

**Comment by stephenroller** [comment]
Combined with #2286, this closes #1876.

File: docs/source/example_seq2seq.rst [hunk 1]
```diff
@@ -1,12 +0,0 @@
...
-  Copyright (c) Facebook, Inc. and its affiliates.
-  This source code is licensed under the MIT license found in the
-  LICENSE file in the root directory of this source tree.
-
-agents.example_seq2seq
-======================
-
-Reference implementation for a simplified Seq2Seq model.
-
-.. automodule:: parlai.agents.examples.seq2seq
```
Comment (dexterju) [inline docs/source/example_seq2seq.rst]: Will this auto generate contents?

Comment (stephenroller) [reply docs/source/example_seq2seq.rst]: Yes https://parl.ai/docs/agent_refs/examples.html

File: docs/source/generate_agent_list.py [hunk 2]
```diff
@@ -3,10 +3,83 @@
...
+
+
+"""
+Extract the readme and CLI args for all of the standard agents.
+"""
+
+
+def _make_argparse_table(class_):
+    """
+    Build the reStructuredText table containing the args and descriptions.
```
Comment (klshuster) [inline docs/source/generate_agent_list.py]: are we inserting an RST table into an markdown file? does that work?

File: docs/source/generate_agent_list.py [hunk 3]
```diff
@@ -3,10 +3,83 @@
...
+            if action.choices:
+                description += [
+                    "Choices: " + ", ".join(f'``{c}``' for c in action.choices) + "."
+                ]
+            # list default and recommended values.
+            default_value = ""
+            if action.default is not None:
+                default_value += f"Default: ``{action.default}``.  "
+            if hasattr(action, 'recommended') and action.recommended:
+                default_value += f"Recommended: ``{action.recommended}``. "
```
Comment (klshuster) [inline docs/source/generate_agent_list.py]: no new line here?

File: parlai/core/params.py [hunk 4]
```diff
@@ -210,6 +210,22 @@ def _format_action_invocation(self, action):
         args_string = self._format_args(action, default)
         return ', '.join(action.option_strings) + ' ' + args_string
 
+    def _get_help_string(self, action):
+        help = action.help
+        if '%(default)' not in action.help:
```
Comment (klshuster) [inline parlai/core/params.py]: what's this `%` doing here?

File: docs/source/generate_agent_list.py (hunk 2 above)
Comment (stephenroller) [reply docs/source/generate_agent_list.py]: uhhhhh yesss. lol

File: parlai/core/params.py (hunk 4 above)
Comment (stephenroller) [reply parlai/core/params.py]: This is the magic constant that is replaced with the default value in argparse. I copy-pasta'd this code directly from Cython source:

https://github.com/python/cpython/blob/5bbac8cbdf140ebce446ea4e7db2b20a5d7b8402/Lib/argparse.py#L691-L698

File: docs/source/generate_agent_list.py (hunk 2 above)
Comment (stephenroller) [reply docs/source/generate_agent_list.py]: I tried to build it in markdown but the markdown plugin doesn't support tables

File: docs/source/generate_agent_list.py (hunk 3 above)
Comment (stephenroller) [reply docs/source/generate_agent_list.py]: No, I wanted to keep them next to each other

**Pull request merged**